packages/errors/src/test-utils/expect-to-throw-fuel-error.ts

```diff
@@ -0,0 +1,47 @@
+import { isAsyncFunction } from 'util/types';
+
+import type { FuelError } from '../index';
+import { ErrorCode } from '../index';
+
+const enumValues = Object.values(ErrorCode);
+
+function assertExpectations(
+  thrownError: FuelError,
+  expectedError: FuelError | (Partial<FuelError> & Required<Pick<FuelError, 'code'>>)
+) {
+  expect(thrownError.name).toEqual('FuelError');
+
+  expect(enumValues).toContain(expectedError.code);
+
+  (Object.keys(expectedError) as Array<keyof typeof expectedError>).forEach((key) => {
+    expect(thrownError?.[key]).toEqual(expectedError[key]);
+  });
+}
+
+export function expectToThrowFuelError<
```

Comment:
With this implementation the `expectToThrowFuelError` function can be used in both a synchronous and asynchronous context in the same way. If you pass an async `lambda`, you'll just have to do `expectToThrowFuelError`. This is the same behavior as doing `await expect(async () => {...}).rejects.toThrow()`, etc.

Reply:
This added a _lot_ of unnecessary complexity to `expectToThrowFuelError` method, and the goal usage deviates from the default way of `expect`-ing async throws in Jest.

Reply:
Please review my [comment](https://github.com/FuelLabs/fuels-ts/pull/1166#discussion_r1285856794) on your PR.